Tests/AppTests/PackageTests.swift

```diff
@@ -354,6 +357,68 @@ final class PackageTests: AppTestCase {
         XCTAssertTrue(pkg.isNew)
     }
 
+    func test_save_platformCompatibility_save() throws {
+        try Package(url: "1".url, platformCompatibility: [.ios, .macos, .ios])
+            .save(on: app.db).wait()
+        let readBack = try XCTUnwrap(Package.query(on: app.db).first().wait())
+        XCTAssertEqual(readBack.platformCompatibility, [.ios, .macos])
+    }
+
+    func test_save_platformCompatibility_read_nonunique() throws {
+        // test reading back of a non-unique array (this shouldn't be
+        // occuring but we can't enforce a set at the DDL level so it's
+        // technically possible and we want to ensure it doesn't cause
+        // errors)
+        try Package(url: "1".url).save(on: app.db).wait()
+        try (app.db as! SQLDatabase).raw(
+            "update packages set platform_compatibility = '{ios,ios}'"
+        ).run().wait()
+        let readBack = try XCTUnwrap(Package.query(on: app.db).first().wait())
+        XCTAssertEqual(readBack.platformCompatibility, [.ios])
+    }
+
+    func test_updatePlatformCompatibility() throws {
+        // setup
+        let p = try savePackage(on: app.db, "1")
+        let v = try Version(package: p, latest: .defaultBranch)
+        try v.save(on: app.db).wait()
+        try Build.Platform.allCases.forEach {
+            // Create a build record for each platform to ensure we can read back
+            // any mapped build correctly.
+            // For instance, if macos-spm wasn't mapped to macos in the update statement,
+            // reading back that package would fail when de-serialising `macos-spm`
+            // into `Package.PlatformCompatibility`, because it has no such enum
+            // case.
+            // We need to test this explicitly, because the raw SQL update statement
+            // in combination with a plain TEXT[] backing field for
+            // platform_compatibility prevents us from relying on type safety.
+            // This test ensures that a newly added case in Build.Platform
+            // must also be handled in the updatePlatformCompatibility SQL
+            // statement.
+            // If it isn't, this test will fail with:
+            // invalid field: platform_compatibility type: Set<PlatformCompatibility> error: Unexpected data type: TEXT. Expected jsonb/json
+            // (which is a bit obscure but means that the content of
+            // platform_compatibility cannot be de-serialised into
+            // PlatformCompatibility)
```

Comment:
This is to deal with the lack of type-safety. We're essentially converting from `PackageShow.Model.PlatformCompatibility` to `Package.PlatformCompatibility` in the update SQL and can't rely on the compiler to check our transformation between the cases. But at least via this test we have that coverage, even if it's not at compile time.